tendrl/common/persistence/persister.py

```diff
@@ -0,0 +1,86 @@
+import logging
+
+import gevent.event
+import gevent.greenlet
+import gevent.queue
+from tendrl.common.etcdobj.etcdobj import Server as etcd_server
+
+
+from tendrl.gluster_integration.config import TendrlConfig
+from tendrl.gluster_integration.persistence.sync_objects import SyncObject
+
+
+LOG = logging.getLogger(__name__)
+
+
+class deferred_call(object):
+
+    def __init__(self, fn, args, kwargs):
+        self.fn = fn
+        self.args = args
+        self.kwargs = kwargs
+
+    def call_it(self):
+        self.fn(*self.args, **self.kwargs)
+
+
+class Persister(gevent.greenlet.Greenlet):
```

Comment:
have a persistor base class (one for persisting to disk and one for persisting to etcd)